networking/ovn_kubernetes_network_provider/ovn-kubernetes-architecture.adoc

```diff
@@ -0,0 +1,51 @@
+:_content-type: ASSEMBLY
+[id="ovn-kubernetes-architecture"]
+= OVN-Kubernetes architecture
+include::_attributes/common-attributes.adoc[]
+:context: ovn-kubernetes-architecture
+
+toc::[]
+
+The following diagram shows the OVN-Kubernetes architecture.
+
+.OVN-Kubernetes architecture
+image::299_OpenShift_OVN-Kubernetes_arch_0123_1.png[OVN-Kubernetes architecture]
+
+The key components are:
+
+* **Cloud Management System (CMS)** - A platform specific client for OVN that provides a CMS specific plugin for OVN integration. The plugin translates the cloud management system's concept of the logical network configuration, stored in the CMS configuration database in a  CMS-specific  format, into an intermediate representation understood by OVN.
+* **OVN Northbound database (`nbdb`)** - Stores the logical network configuration passed by the CMS plugin.
+* **OVN Southbound database (`sbdb`)** - Stores the physical and logical network configuration state for OpenVswitch (OVS) system on each node, including tables that bind them.
+* **ovn-northd** - This is the intermediary client between `nbdb` and `sbdb`. It translates  the logical network configuration in terms of conventional network concepts, taken from the `nbdb`, into  logical data path flows in the `sbdb` below it. The container name is `northd` and it runs in the `ovnkube-master` pods.
+* **ovn-controller** - This is the OVN agent that interacts with OVS and hypervisors, for any information or update that is needed for `sbdb`. The `ovn-controller` reads logical flows from the `sbdb`, translates them into `OpenFlow` flows and sends them to the node’s OVS daemon. The container name is `ovn-controller` and it runs in the `ovnkube-node` pods.
+
+The OVN northbound database has the logical network configuration passed down to it by the cloud management system (CMS).
+The OVN northbound Database contains the current desired state of the network, presented as a collection of logical ports, logical switches, logical routers, and more.
+The `ovn-northd` (`northd` container) connects to the OVN northbound database and the OVN southbound database.
+It translates the logical network configuration in terms of conventional network concepts, taken from the OVN northbound Database, into logical data path flows in the OVN southbound database.
+
+The OVN southbound database has physical and logical representations of the network and binding tables that link them together. Every node in the cluster is represented in the southbound database, and you can see the ports that are connected to it.
+It also contains all the logic flows, the logic flows are shared with the `ovn-controller` process that runs on each node and the `ovn-controller` turns those into `OpenFlow` rules to program `Open vSwitch`.
```

Comment:
This information should exist in a concept module. Maybe a reference.

Reply:
done